dynacred/src/Contact.ts

```diff
@@ -349,20 +392,20 @@ export class Contact {
             return Log.rcatch(e, "couldn't convert to Contact from UserLocation");
         }
     }
-    credentialInstance: CredentialsInstance = null;
-    darcInstance: DarcInstance = null;
-    coinInstance: CoinInstance = null;
-    spawnerInstance: SpawnerInstance = null;
-    recover: Recover = null;
-    calypso: Calypso = null;
-    bc: ByzCoinRPC = null;
-    contactsCache: Contact[] = null;
-    actionsCache: DarcInstance[] = null;
-    groupsCache: DarcInstance[] = null;
-
-    constructor(public credential: CredentialStruct = null,
-                public data: Data = null) {
-        if (credential == null) {
+    credentialInstance: CredentialsInstance = undefined;
+    darcInstance: DarcInstance = undefined;
+    coinInstance: CoinInstance = undefined;
+    spawnerInstance: SpawnerInstance = undefined;
+    recover: Recover = undefined;
+    calypso: Calypso = undefined;
+    bc: ByzCoinRPC = undefined;
+    contactsCache: Contact[] = undefined;
+    actionsCache: DarcInstance[] = undefined;
+    groupsCache: DarcInstance[] = undefined;
```

Comment:
You don't need to explicit `undefined` assignation, but you can correct the types to be also undefined.

Reply:
Except for `credential`, `recover` and `calypso` as these are definitly assigned in the constructor.